File: src/main/java/mods/eln/sim/thermal/process/RegulatorFurnaceProcess.java
package mods.eln.sim.thermal.process;

import mods.eln.sim.RegulatorProcess;

public class RegulatorFurnaceProcess extends RegulatorProcess {

    FurnaceProcess furnace;

    public RegulatorFurnaceProcess(String name, FurnaceProcess furnace) {
        super(name);
        this.furnace = furnace;
    }

    @Override
    protected double getHit() {
        return furnace.load.Tc;
    }

    @Override
    protected void setCmd(double cmd) {
        furnace.setGain(cmd);
    }
}

File: src/main/java/mods/eln/sim/RegulatorFurnaceProcess.kt
package mods.eln.sim

import mods.eln.sim.thermal.FurnaceProcess

open class RegulatorFurnaceProcess(name: String, internal var furnace: FurnaceProcess) : RegulatorProcess(name) {

    override fun getHit(): Double {
        return furnace.load.Tc
    }

    override fun setCmd(cmd: Double) {
        furnace.gain = cmd
    }
}
